projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -762,7 +768,14 @@ async def _generate_embeddings(
         documents: list[str] = []
         for item in items:
             pk = self._get_item_pk_values(item)
-            chunks = self.vectorizer.config.chunking.into_chunks(item)
+            if self.vectorizer.config.loader is not None:
+                file_content = self.vectorizer.config.loader.load(item)
+                markdown_content = self.vectorizer.config.parser.parse_file_content(
+                    file_content
+                )
+                chunks = self.vectorizer.config.chunking.into_chunks(markdown_content)
+            else:
+                chunks = self.vectorizer.config.chunking.into_chunks(item)
             for chunk_id, chunk in enumerate(chunks, 0):
                 formatted = self.vectorizer.config.formatting.format(chunk, item)
```

Comment:
I think it may be best to conceive of this as a processing pipeline where most processing nodes alter the row

```suggestion
            if self.vectorizer.config.loader is not None:
                # loader reads a "url" column and loads binary data into item['file_contents']
                self.vectorizer.config.loader.load(item)
            if self.vectorizer.config.parser is not None:
               # reads some column (default: file_contents) and puts result in item['parsed_file_content']
               self.vectorizer.config.parser.parse_file_content(item)
            chunks = self.vectorizer.config.chunking.into_chunks(item)
            for chunk_id, chunk in enumerate(chunks, 0):
                formatted = self.vectorizer.config.formatting.format(chunk, item)
                records_without_embeddings.append(pk + [chunk_id, formatted])
```